### PR TITLE
Plugin Uploads: Add a plugin upload feature to checkout thank you page

### LINKED ISSUE #15673: Plugin Uploads: Add plugin upload button to "thank you" page after plan purchase
After purchase of a business plan via the nudge from #15668 and the plans page in #15671, make it easy to upload a plugin by showing a button on the thank you page:

screen shot 2017-06-29 at 17 27 46

### PR DESCRIPTION
**Pull request by tyxla**
This PR adds a "Upload a Plugin" feature to the "Thank You" post checkout page for users who just bought a Business plan.

Before:

After:

Fixes #15673

To test:
* Checkout this branch
* Purchase a Business plan for one of your .com sites.
* Verify you can see the plugin upload feature in the thank you page.
* Verify the plugin upload feature takes you to the plugin upload page (even though it's disabled for now).

In addition to a code and design review, this could use a copy review, cc @Automattic/editorial

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx [hunk 1]
```diff
@@ -46,6 +46,17 @@ const BusinessPlanDetails = ( { selectedSite, sitePlans, selectedFeature } ) =>
...
 			}
 
+			{ ! selectedFeature &&
+				<PurchaseDetail
+					icon="plugins"
+					title={ i18n.translate( 'Upload a Plugin' ) }
+					description={ i18n.translate(
+						'You can now upload your own plugins, or ones you\'ve downloaded, directly through a drag and drop interface.'
+					) }
+					buttonText={ i18n.translate( 'Upload a plugin now' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx]: Hi! I've found a possible matching string that has already been translated **68** times:
`translate( 'Upload Plugin' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx [hunk 2]
```diff
@@ -46,6 +46,17 @@ const BusinessPlanDetails = ( { selectedSite, sitePlans, selectedFeature } ) =>
 					href={ '/themes/' + selectedSite.slug } />
 			}
 
+			{ ! selectedFeature &&
+				<PurchaseDetail
+					icon="plugins"
+					title={ i18n.translate( 'Upload a Plugin' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx]: Hi! I've found a possible matching string that has already been translated **68** times:
`translate( 'Upload Plugin' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by tyxla** [comment]
#17243 addresses this one, but for the My Plan page.

**Comment by rickybanister** [comment]
The design in the issue does not match the design in this PR—is the design from the issue the new (not yet implemented version?)

**Comment by rickybanister** [comment]
If that's the case then this looks good.

**Comment by tyxla** [comment]
Yup, that's exactly the case @rickybanister.

File: client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx [hunk 3]
```diff
@@ -46,6 +47,17 @@ const BusinessPlanDetails = ( { selectedSite, sitePlans, selectedFeature } ) =>
 					href={ '/themes/' + selectedSite.slug } />
 			}
 
+			{ ! selectedFeature && isEnabled( 'manage/plugins/upload' ) &&
+				<PurchaseDetail
+					icon="plugins"
+					title={ i18n.translate( 'Upload a Plugin' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx]: Hi! I've found a possible matching string that has already been translated **68** times:
`translate( 'Upload Plugin' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by michelleweber** [comment]
I'd make a couple of tweaks:

Add a Plugin
Search and add plugins right from your dashboard, or upload a plugin from your computer with a drag-and-drop interface.

Also -- if I can make a couple of suggestions for the other bits, while changes are being made to this screen:

Find a new theme
You've now got access to every premium theme, at no extra cost -- that's hundreds of new options. Give one a try!

Connect to Google Analytics
Complement WordPress.com's stats with Google's in-depth look at your visitors and traffic patterns.

**Comment by tyxla** [comment]
Thanks so much for the copy suggestions @michelleweber, and for the review @ockham 👍 

I've updated the copy and will handle the other suggestions in separate PRs.

File: client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx [hunk 4]
```diff
@@ -46,6 +47,18 @@ const BusinessPlanDetails = ( { selectedSite, sitePlans, selectedFeature } ) =>
...
 
+			{ ! selectedFeature && isEnabled( 'manage/plugins/upload' ) &&
+				<PurchaseDetail
+					icon="plugins"
+					title={ i18n.translate( 'Add a Plugin' ) }
+					description={ i18n.translate(
+						'Search and add plugins right from your dashboard, or upload a plugin ' +
+						'from your computer with a drag-and-drop interface.'
+					) }
+					buttonText={ i18n.translate( 'Upload a plugin now' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/business-plan-details.jsx]: Hi! I've found a possible matching string that has already been translated **68** times:
`translate( 'Upload Plugin' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by tyxla** [comment]
I've addressed the theme upload copy updates in #17267, and the ones for Google Analytics in #17268 - they could use a quick sanity check before i 🚢 .

**Comment by sirreal** [comment]
This is _probably_ an unrelated bug, but I'm bringing it up here as I uncovered it while testing.

When I land on the "Thank You" page, if I quickly click on the "Upload a plugin now" button, I see the following confirm dialog:

> We haven't finished installing your plugins.

weird

This is the same confirm dialog you see on the new Jetpack Thank You page.

/cc @oskosk

**Comment by tyxla** [comment]
Seems like this can happen only if you've purchased a Jetpack plan for one of your Jetpack sites prior to your WP.com plan purchase. This seems to happen because we're not cleaning up the `page.exit` route mapping when we switch the site: https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx#L130

I must admit this is quite an edge case, but it's worth reporting.

Finally, this isn't related with the PR at all, so we should be safe to land it.

**Comment by rachelmcr** [comment]
@tyxla Ready to merge this one?

**Comment by tyxla** [comment]
@rachelmcr Yes, I was actually waiting for a final thumbs up from @seear, who returns today from vacation.

**Comment by seear** [comment]
Excellent, looks ready to go to me 👍

**Pull request merged**